docs/advanced/peering/peering-via-cr.md

```diff
@@ -0,0 +1,428 @@
+pod-cidr        10.244.0.0/16   10.244.0.0/16   3h16m
+service-cidr    10.96.0.0/16    10.96.0.0/16    3h16m
+```
+
+### Creating and exchanging the Wireguard keys (both clusters)
+
+To enable authentication and encryption, Wireguard requires a key pair, one for each gateway.
+
+Those keys can be created [via the Wireguard utility tool](https://www.wireguard.com/quickstart/#key-generation) or via OpenSSL like so:
+
+```{code} bash
+openssl genpkey -algorithm X25519 -outform der -out private.der
+openssl pkey -inform der -in private.der -pubout -outform der -out public.der
+
+# Get the Wireguard private key
+echo "Private key:"; cat private.der | tail -c 32 | base64
+# Get the Wireguard public key
+echo "Public key:"; cat public.der | tail -c 32 | base64
+```
+
+At this point, **in each cluster**, we will need to create a secret containing this pair of keys:
+
+```{code} yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    liqo.io/remote-cluster-id: <REMOTE_CLUSTER_ID>
+  name: gw-keys
+  namespace: <TENANT_NAMESPACE>
+type: Opaque
+data:
+  privateKey: <WIREGUARD_PRIVATE_KEY>
+  publicKey: <WIREGUARD_PUBLIC_KEY>
+```
+
+where WIREGUARD_PRIVATE_KEY and WIREGUARD_PUBLIC_KEY are the previously generated key pairs for that specific cluster (**each cluster has its pair of keys**).
+
+Additionally, each cluster should have a PublicKey resource **with the public key of the peer cluster** (cluster A has the public key of B and vice versa):
+
+```{code} yaml
+apiVersion: networking.liqo.io/v1beta1
+kind: PublicKey
+metadata:
+  labels:
+    liqo.io/remote-cluster-id: <HERE_THE_CLUSTER_ID_OF_PEER_CLUSTER>
+    networking.liqo.io/gateway-resource: "true"
+  name: gw-publickey
+  namespace: <TENANT_NAMESPACE>
+spec:
+  publicKey: <REMOTE_WIREGUARD_PUBLIC_KEY>
+```
+
+In order to make things work, make sure that the PublicKey resource has the labels:
+
+```{code} yaml
+liqo.io/remote-cluster-id: <HERE_THE_CLUSTER_ID_OF_PEER_CLUSTER>
+networking.liqo.io/gateway-resource: "true"
+```
+
+### Configuring the server gateway (cluster provider)
+
+As mentioned before, a Wireguard tunnel connects the clusters peered with Liqo.
+In this section we are going to configure the gateway server, where the client will connect to.
+
+The `GatewayServer` resource describes the configuration of the gateway server, and should be applied on the cluster acting as server for the tunnel creation.
+You can check [here](./inter-cluster-network.md#creation-of-a-gateway-server) an example of the `GatewayServer` CR.
+
+When you create the `GatewayServer` resource, **make sure to specify the `secretRef`** pointing to the key pairs we created before.
+
+Note that under `.spec.endpoint` of the `GatewayServer` resource you can configure a fixed `nodePort` or `loadBalancerIP` (if supported by your provider) to have a deterministic port or ip address for the gateway, so that the configuration of the client that connects to it can be defined in advance.
```

Comment:
```suggestion
Note that under `.spec.endpoint` of the `GatewayServer` resource you can configure a fixed `nodePort` or `loadBalancerIP` (if supported by your provider) to have a precise UDP port or IP address for the gateway, so that the configuration of the client that connects to it can be defined in advance.
```